### ISSUE TITLE
JSON error doesn't allow me to open VS code from a terminal prompt

### ISSUE THREAD

**Issue by jenslisinski** [Microsoft/vscode#25638]
Getting this error when trying to open VS code from my terminal prompt:

![image](https://cloud.githubusercontent.com/assets/28154049/25541849/43bf16f4-2c1e-11e7-88b8-423f15db8e60.png)

And I get this error message (at top) when I open VS code through the Applications window:

![image](https://cloud.githubusercontent.com/assets/28154049/25541894/694244be-2c1e-11e7-97f6-25deeeefabbc.png)

This is being done from a Mac running Sierra 10.12.4.  Any ideas?  Thanks!

**Comment by alexandrudima** [comment]
@jenslisinski
What version of VS Code are you using?

@dbaeumer 
The only JSON.parse I could find in bootstrap-amd.js executing statically (i.e. within the `Module._compile`) was:

```
    rawNlsConfig = process.env.VSCODE_NLS_CONFIG,
    nlsConfig = rawNlsConfig ? JSON.parse(rawNlsConfig) : {
        availableLanguages: {}
    };
```

**Comment by dbaeumer** [comment]
The interesting question here is why the `VSCODE_NLS_CONFIG` env variable which we control has a broken JSON in there. We set this env variable in the main process.

@jenslisinski could you do the following when this happens:

- open dev tools like you did 
- in the console execute `process.env["VSCODE_NLS_CONFIG"]`
- provide the output here

**Comment by dbaeumer** [comment]
@joaomoreno could this be a problem that we corrupt the env when transferring it from the main to the renderer, extension host

**Comment by joaomoreno** [comment]
Corruption is unlikely...

**Comment by jenslisinski** [comment]
Sorry, everyone, been away for a long weekend in the meantime.

@alexandrudima, I am running VSCode 1.11.2

@dbaeumer, here's the output you requested:  "{locale:en-us,availableLanguages:{}}"

**Comment by dbaeumer** [comment]
Actually that is not the correct content: It should be `"{"locale":"en","availableLanguages":{}}"` (observe all the quotes are missing around the property names).

@joaomoreno it looks like that value is corrupted. Any idea how this can happen

**Comment by jenslisinski** [comment]
Or is there simply a way to fix it, so I can get rid of this behavior and error message (not detrimental to my use, but annoying in that I can't see any of my tabs while editing)

**Comment by dbaeumer** [comment]
@jenslisinski the problem is that we have no real idea yet why this is happening and it doesn't reproduce on any of our machines. Do you use a special shell ?

**Comment by dbaeumer** [comment]
I could catch the exception and fall back to English, however this will only hide the problem and if data in the env get corrupted then we might fail later on as well. So I still think we should try to understand why this is happening.

**Comment by jenslisinski** [comment]
ok
I run this from a Mac terminal session, and I believe it's using bash

**Comment by dbaeumer** [comment]
@jenslisinski just and idea: could it be that the variable is already defined in our env. If you open a terminal and you do echo $VSCODE_NLS_CONFIG does it print anything ?

**Comment by jenslisinski** [comment]
@dbaeumer, yes!  It shows {locale:en-us,availableLanguages:{}}

**Comment by dbaeumer** [comment]
@jenslisinski just to be 100% clear: you see `{locale:en-us,availableLanguages:{}}` when you open a normal terminal (no VS Code) open.

If so can you look into your .bashrc whether it defines such an env variable. If so delete it. Would be interesting to know how it got there in the first place.

**Comment by jenslisinski** [comment]
@dbaeumer  Done.  And that seems to have cleared up the problem.

It was in my .bashrc_<machine code> file (specific to my Mac). I've never been in this file prior to today, so I can only think it got there when I installed VSCode thru homebrew?  It's also mysterious as to why this behavior suddenly started happening after I had used VSCode for a while.  Presumably, that env variable was there all along.

Thank you for your help!
